examples/api/src/app/api/erc-20/route.ts

```diff
@@ -0,0 +1,78 @@
+import { NextRequest } from "next/server";
+import {
+  getFollowingHolderInfo,
+  getPriceData,
+  parseTokenParam,
+  getTokenInfo,
+} from "./lib/utils";
+
+export async function GET(request: NextRequest) {
```

Comment:
this whole function should be 3 separate endpoints - this abstraction is fairly complex to reason about